core/src/main/scala/tapir/SchemaFor.scala

```diff
@@ -60,7 +60,13 @@ object SchemaFor extends SchemaForMagnoliaDerivation {
     override def schema: Schema = implicitly[SchemaFor[T]].schema
   }
 
-  implicit def schemaForMap[V: SchemaFor]: SchemaFor[Map[String, V]] = SchemaFor(SProduct(SObjectInfo("Map"), List.empty, List.empty))
+  implicit def schemaForMap[V: SchemaFor]: SchemaFor[Map[String, V]] = {
+    val additionalProperties = implicitly[SchemaFor[V]].schema match {
+      case s: SObject => Some(s)
+      case _          => None
```

Comment:
shouldn't `additionalProperties` be always specified, as we always know the schema of the keys? see the string-string example here: https://swagger.io/docs/specification/data-models/dictionaries/